src/App.tsx

```diff
@@ -1,41 +1,109 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { getPosts, getUserPosts, selectedPostId } from './api/posts';
+import { getUsers } from './api/users';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
-
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
-
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
-
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+const App: React.FC = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [postDetails, setPostDetails] = useState<Post | null>(null);
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectUser, setSelectUser] = useState(0);
+  const [selectPost, setSelectPost] = useState(0);
+
+  const loadPosts = async () => {
+    const postsFromServer = await getPosts();
+
+    setPosts(postsFromServer);
+  };
+
+  const loadSelectedPosts = async () => {
+    const postsFromServer = await getUserPosts(selectUser);
+
+    setPosts(postsFromServer);
+  };
+
+  const loadUsers = async () => {
+    const usersFromServer = await getUsers();
+
+    setUsers(usersFromServer);
+  };
+
+  const loadPostDetails = async (id: number) => {
+    if (selectPost !== 0) {
+      const post = await selectedPostId(id);
+
+      setPostDetails(post);
+    }
+  };
+
+  useEffect(() => {
+    loadPosts();
+    loadUsers();
+  }, []);
+
+  useEffect(() => {
+    if (selectUser !== 0) {
+      loadSelectedPosts();
+    } else {
+      loadPosts();
+    }
+  }, [selectUser]);
+
+  useEffect(() => {
+    loadPostDetails(selectPost);
+  }, [selectPost]);
+
+  const selectHendler = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectUser(+event.target.value);
+  };
+
+  const openDetailsHendler = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setSelectPost(+event.currentTarget.value);
+  };
```

Comment:
Read more about function-handler naming: https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364